### PR TITLE
PRT-189: Update exampe-implementations to use yarn

### PR DESCRIPTION
**Pull request by lazaralex98**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,7 +3,7 @@ node_modules
 coverage
 coverage.json
 typechain
-yarn.lock
+package-lock.json
```
Comment (kargakis) [inline .gitignore:6]: `package-lock.json` should be removed from gitignore so if it is generated by mistake for anyone locally, it is easy to spot.

Comment (lazaralex98) [reply .gitignore:6]: makes sense, will do

**Comment by kargakis** [comment]
You will need to rebase this PR.

File: hardhat.config.ts [hunk 2]
```diff
@@ -47,7 +47,8 @@ const config: HardhatUserConfig = {
     hardhat: {
       forking: {
         url:
-          process.env.POLYGON_URL || "",
+          process.env.POLYGON_URL ||
+          "https://polygon-mainnet.g.alchemy.com/v2/4rzRS2MH5LIunV6cejmLhQelv_Vd82rq",
         blockNumber: 25848203,
```
Comment (kargakis) [inline hardhat.config.ts:52]: Should we just drop the blockNumber here so we always test against the latest version of our contracts?

Comment (lazaralex98) [reply hardhat.config.ts:52]: There's one main reason why I'm against that: token balances of impersonated addresses can change and mess with the tests.

As a nice side note, pinning one blockNumber also allows hardhat to cache the blockchain tests => faster tests.

Comment (kargakis) [reply hardhat.config.ts]: If we use hardhat's account, then the account always has a balance of 10000 ETH as part of the fork. Also caching doesn't help in CI but only if you develop locally. I think these tests are not going to need a lot of development and utilizing the CI as a canary to inform us whether our examples work against latest core is a major advantage.

Comment (lazaralex98) [reply hardhat.config.ts]: You do have a point that it's super useful to test against the latest core.

But the hardhat account having 10000 ETH doesn't help. Some of the tests have the user deposit NCT or BCT into the contract.

Is there a way to have hardhat give an account tokens out of thin air in a fork? I tried and couldn't find anything like that.

Comment (kargakis) [reply hardhat.config.ts]: Yes, you should be able to top up other accounts with [`myAddress`](https://github.com/ToucanProtocol/example-implementations/blob/3969ca89598d9e5c3e7c5fce63ee96948f4907e7/utils/addresses.ts#L11) which is the first hardhat account.

Comment (lazaralex98) [reply hardhat.config.ts]: So, just buy BCT & NCT on my address. I guess it was so simple I didn't think of it 🤦🏻‍♂️🤣

Comment (kargakis) [reply hardhat.config.ts]: Exactly, create a beforeEach where you swap from MATIC to BCT/NCT then top up other accounts with these tokens.

Comment (lazaralex98) [reply hardhat.config.ts]: Let's merge this and I'll open a different PR to remove the blockNumber and edit tests to use `myAddress`. cool?

**Review by kargakis — APPROVED** [review]

**Pull request merged**